### ISSUE TITLE
Problems with composer update

### ISSUE THREAD

**Issue by pwronline** [appstract/laravel-options#14]
Hi,
I'm having this problem when updating with composer update with composer install no problem.
Thxs!

Declaration of Appstract\Options\Option::delete($key) should be compatible with Illuminate\Database\Eloquent\Model::delete()

**I have continued to investigate, the error appears when I updated your package to 1.1.0**

```
"require": {
        "php": "^7.1.3",
        "appstract/laravel-options": "^1.0",
        "fideloper/proxy": "^4.0",
        "intervention/image": "^2.4",
        "laravel/framework": "5.7.*",
        "laravel/tinker": "^1.0",
        "laraveles/spanish": "^1.1",
        "spatie/laravel-permission": "^2.26",
        "yajra/laravel-datatables-oracle": "~8.0"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.2",
        "beyondcode/laravel-dump-server": "^1.0",
        "filp/whoops": "^2.0",
        "fzaninotto/faker": "^1.4",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^2.0",
        "phpunit/phpunit": "^7.0"
    },
```

**Comment by teuunn** [comment]
Hi @pwronline,

I can't reproduce this issue when:

1. New Laravel 5.7.* project
2. Composer require appstract/laravel-options
3. Composer update

I expected your error in step 3, but it didn't show up. Maybe it's a conflict with one of the other packages?

**Comment by htulibacki** [comment]
Hi @teuunn, 

your \Appstract\Option::delete method should be compatible with \Illuminate\Database\Eloquent\Model::delete (yours have an $key attribute). It's not a good idea to override Eloquents methods - use model events instead.

**Comment by wassim** [comment]
@teuunn Getting this error too with 1.1.0

**Comment by teuunn** [comment]
Hi @htulibacki 

Definitely the way I would take when building something new. For now maybe a quick fix can eliminate the issue

**Comment by teuunn** [comment]
Hi @wassim, do you have an example to reproduce the issue?

PR's are welcome too.

**Comment by MarkVaughn** [comment]
@teuunn https://github.com/appstract/laravel-options/pull/15 addresses the issue (changing delete -> deleteKey)
the issue was introduced here https://github.com/appstract/laravel-options/pull/13

**Comment by gizburdt** [comment]
I think we need to think about all method names then. It's not really consistent if the delete method is called deleteKey (with -Key) and all other methods don't have the -Key in the method name.

Maybe an idea to rename the delete method to remove or destroy?

**Comment by MarkVaughn** [comment]
changed it to remove @gizburdt

**Comment by gizburdt** [comment]
Thanks, will have a look at later :)

**Comment by gizburdt** [comment]
Merged and tagged v2.0.0
